application/mounts.py

```diff
@@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
 		mount_uri = mount.get_root().get_uri()
 		volume = mount.get_volume()
 
+		# check if mount has been added as volume already
+		mount_already_in_volume_list = False
```

Comment:
Hi Mladen,

Compared to GIO directly mounting a SD card (where Sunflower just 
received a single "add mount" event), there are three events when my 
Android phone gets mounted:

- "add volume" event with a volume object that has the mount correctly set
- first "add mount" event with a passed mount argument with volume being 
None on it
- second "add mount" event with a passed mount argument with volume set

With the code, I wanted to make sure we don't end up with two mounts 
displayed in the "Mount Manager" window and bookmark list.

This is on a plain Ubuntu 14.04.

Does this make any sense?

Christian

On 15/05/15 20:19, Mladen Mijatov wrote:
>
> In application/mounts.py 
> <https://github.com/MeanEYE/Sunflower/pull/112#discussion_r30437718>:
>
> > @@ -42,11 +42,21 @@ def _add_mount(self, monitor, mount):
> >  		mount_uri = mount.get_root().get_uri()
> >  		volume = mount.get_volume()
> >
> > +		# check if mount has been added as volume already
> > +		mount_already_in_volume_list = False
>
> Why do you have this check? It's suppose to be managed already.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/MeanEYE/Sunflower/pull/112/files#r30437718>.
>